### ISSUE TITLE
Function for creating Affine transforms

### ISSUE THREAD

**Issue by nexplor** [sebcrozet/nalgebra#364]
It would be useful to have (fairly straightforward) functions Affine2::new and Affine3::new (or similar names), that take an (n x (n+1)) matrix, and return an affine transform (by appending the row (0, 0, ... 0, 1)). This prevents the necessity to call from_matrix_unchecked each time one wants an affine transform to be created.

**Comment by dragostis** [comment]
I may not be understanding everything this to the full extent, but even though projectives contain affines and affines contain similarities, it's hard to convert between them. How does one create a similarity, then add a non-uniform scaling to it by converting it to an affine, for example?

**Comment by sebcrozet** [comment]
@nexplor Yes, this would be useful. Though the constructor would still have to check that the matrix is invertible.

@dragostis 
> How does one create a similarity, then add a non-uniform scaling to it by converting it to an affine, for example?

Since the `Affine` transformation is stored as a matrix in homogeneous coordinates you can do, for example: `Affine3::from_matrix_unchecked(similarity.to_homogeneous())`. Though I agree a more explicit conversion method would be useful to avoid the `from_matrix_unchecked`.

> Would there be a way for these traits to have some inheritance implemented within them? So that, if I apply a non-uniform scale and a translation to the identity, I can choose between AffineTransformations and ProjectedTransformations and anything else higher up the inheritance ladder. Also, having all the lower methods available in higher contexts would be useful.

I'm not sure what you mean by inheritance here. I you are referring to  inheritance between alga traits, this is already the case.
